### DISCUSSION TITLE
Wemos S2 Mini not supported anymore

### DISCUSSION THREAD

**Discussion by Robinyoh** [arendst/Tasmota#18314] [Q&A] [answered]
### PROBLEM DESCRIPTION:

Purple Wemos S2 Mini was supported in some firmware back in the days, I was succesfull to flash firmware by webinstaller. But now after flashing the wifi manager is not present anymore after uploading.
Is this for a reason?

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): _____
- [x] Tasmota binary firmware version number used: _____
  - [x] Pre-compiled
  - [x] Self-compiled
- [x] Flashing tools used: _____
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [x] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [x] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by barbudor** [comment, marked as answer]
S2 is supported whatever the board and it's colour
You probably used a bad power supply when flashing the device

**Comment by Robinyoh** [comment]
Just using the usb-c connector of my MacBook Pro. Uploader by Arduino IDE works flawless. And it was working before, but I don't know the firmware at the time. It was somewhere begin August 2022 because I orders it late September.

**Reply by sfromis** [reply]
I'm running Wemos S2 Mini with current Tasmota without experiencing connection problems. Note that I'm using the original branded board, not some clone, and I've seen at least one case of a user having significant trouble making a clone board work.

**Comment by Robinyoh** [comment]
I have a clone indeed, from Ali. But I don't understand why it was working before? Did something change? Do you guys are going to support clone esp32 s2 mini or is it just the build of it?

The same esp32 s2 mini clone was working before, after using it for an Arduino IDE project I try to flash it again with tasmota without succes anymore. I have the one with ram not the one without.

**Reply by sfromis** [reply]
Tasmota supports the ESP32-S2 chip variant. Whether some arbitrary board works depends on the hardware, not on Tasmota, and Tasmota does not even try to adapt to what your board happens to be.

In case of a bad hardware design, things like power issues can screw up things. This is not only about what power you supply, but about the on-board voltage regulator, and also whether there are suitable capacitors where they should be. The user I told about was able to get it on Wifi while hand-holding the board, but not when letting it go. And of course if there are faults on a specific board, but bad China-clones are a pest.

Things going from working to not is a complex issue, but may be up to the hardware and "quality". Minor differences in power draw can also throw off borderline designs.  Things may also depend a lot on what you have connected to the board.

**Reply by Lassska** [reply]
After install tasmota and reset S2 mini, are you setting the ssid/pass? If not... it's in ap mode (you will find a wifi network on the phone or PC with the name tasmota-xxxxxx-yyyy; 
I associate you with it preferably from the PC and the browser will automatically open at 192.168.4.1 where you can set wifi now)

Tasmota
Successful WiFi Connection

Redirecting to new device's IP address

[192.168........](http://192.168........./)
[Tasmota 13.1.0 by Theo Arends](https://bit.ly/tasmota)

**Reply by Robinyoh** [reply]
Thanks @Lassska Didn't noticed that it created an AP. Was in the assumption that the browser should redirect me to the device to set the wifi credentials like before over the serial port.

**Comment by Jibun-no-Kage** [comment]
I have seen a similar issue, in that some of my older Wemos D1 modules worked fine with firmware 9.3.1 to 12.x, but with 13.x I stated seeing the boards fail to connect to any access point.  Moreover, when flashed with 13.x I could not consistently connect to the boards via default AP at 192.168.4.1.  My phone would never connect, which I can use under 9.3.1 to 12.x without issue.  My PC would sometimes, rarely connect to 192.168.4.1... but it was unstable.  If I take the same board, and flash back to 9.3.1 (to 12.x) it is rock solid wifi connection!  So something planned or unplanned is impacting the consistency of the wifi on some older modules with 13.x releases.

**Reply by sfromis** [reply]
As already suggested in [the other thread](https://github.com/arendst/Tasmota/discussions/19827), it is plausible that your troubles are a power-related issue, not a Tasmota issue. And Tasmota is likely to draw a bit more power in version 13.2 than in 9.3.1.

**Comment by lalo-uy** [comment]
Did you try the same board with 13.1 powered by a better 3.3v PSU?

**Comment by Jibun-no-Kage** [comment]
Yeah... I was thinking that too... but I was using 5v 4a stepped down to 3.3v for flashing, then used for testing once flashed.  Of course, this issue I am seeing, could be the age of the modules as well, they are older, not used much, but older.  The break point in firmware versions is 13.x, if I stay below 13.x, no issues, but once I use anything 13.x, I see the issue.

I left one module up, letting it loop...

00:00:10.532 WIF: Connect failed as AP cannot be reached
00:00:10.870 WIF: Connecting to AP1 DD in mode 11n as tasmota-8FF589-5513...
00:00:25.023 WIF: Connect failed with AP timeout
00:00:26.376 WIF: Connecting to AP1 DD in mode 11n as tasmota-8FF589-5513...
00:00:42.011 WIF: Connect failed with AP timeout
00:00:42.349 WIF: Connecting to AP1 DD in mode 11n as tasmota-8FF589-5513...
00:00:57.012 WIF: Connect failed with AP timeout
00:00:58.351 WIF: Connecting to AP1 DD in mode 11n as tasmota-8FF589-5513...
00:01:14.017 WIF: Connect failed with AP timeout
00:01:14.355 WIF: Connecting to AP1 DD in mode 11n as tasmota-8FF589-5513...
00:01:28.013 WIF: Connect failed as AP cannot be reached
00:01:29.013 WIF: Connect failed as AP cannot be reached
00:01:29.351 WIF: Connecting to AP1 DD in mode 11n as tasmota-8FF589-5513...
00:01:43.027 WIF: Connect failed as AP cannot be reached
00:01:44.027 WIF: Connect failed as AP cannot be reached
00:01:44.365 WIF: Connecting to AP1 DD in mode 11n as tasmota-8FF589-5513...
00:01:59.029 WIF: Connect failed with AP timeout
00:02:00.370 WIF: Connecting to AP1 DD in mode 11n as tasmota-8FF589-5513...
00:02:16.039 WIF: Connect failed with AP timeout
00:02:16.377 WIF: Connecting to AP1 DD in mode 11n as tasmota-8FF589-5513...
00:02:31.018 WIF: Connect failed with AP timeout
00:02:32.358 WIF: Connecting to AP1 DD in mode 11n as tasmota-8FF589-5513...
00:02:48.028 WIF: Connect failed with AP timeout
00:02:48.366 WIF: Connecting to AP1 DD in mode 11n as tasmota-8FF589-5513...
00:03:03.024 WIF: Connect failed with AP timeout
00:03:04.365 WIF: Connecting to AP1 DD in mode 11n as tasmota-8FF589-5513...
00:03:20.012 WIF: Connect failed with AP timeout
00:03:20.360 WIF: Connecting to AP1 DD in mode 11n as tasmota-8FF589-5513...
00:03:29.261 WIF: Connect failed as AP cannot be reached
00:03:30.262 WIF: Connect failed as AP cannot be reached
00:03:30.600 WIF: Connecting to AP1 DD in mode 11n as tasmota-8FF589-5513...
00:03:45.032 WIF: Connect failed with AP timeout
00:03:46.370 WIF: Connecting to AP1 DD in mode 11n as tasmota-8FF589-5513...
00:04:02.019 WIF: Connect failed with AP timeout
00:04:02.357 WIF: Connecting to AP1 DD in mode 11n as tasmota-8FF589-5513...
00:04:17.013 WIF: Connect failed with AP timeout
00:04:18.362 WIF: Connecting to AP1 DD in mode 11n as tasmota-8FF589-5513...
00:04:34.016 WIF: Connect failed with AP timeout
00:04:34.355 WIF: Connecting to AP1 DD in mode 11n as tasmota-8FF589-5513...
00:04:44.019 WIF: Connect failed
00:04:45.024 WIF: Connect failed
00:04:45.361 WIF: Connecting to AP1 DD in mode 11n as tasmota-8FF589-5513...
00:05:00.024 WIF: Connect failed with AP timeout
00:05:01.371 WIF: Connecting to AP1 DD in mode 11n as tasmota-8FF589-5513...
00:05:16.015 WIF: Connect failed
00:05:16.353 WIF: Connecting to AP1 DD in mode 11n as tasmota-8FF589-5513...
00:05:31.010 WIF: Connect failed with AP timeout
00:05:32.349 WIF: Connecting to AP1 DD in mode 11n as tasmota-8FF589-5513...
00:05:43.013 WIF: Connected

I checked my DHCP server... restarted it, etc., so don't think that was a factor.

So this may even be an SDK issue?  I seem to recall a old thread here, about a similar problem where it was believed the SDK had a quirk or the retained wifi configuration was tripping things up... I just don't recall the thread # at the moment.

For now, I can live with 12.x firmware which is stable on these older modules I have.  When I can, I will test more.  At this point this issue is more 'interesting' then 'frustrating' for me. :)  I am tempted to pull out the Arduino IDE or MicroPython and see what happens.

As always, thanks for the active support, greatly appreciated.

**Reply by sfromis** [reply]
Better not discuss your troubles here, when you already have a thread about it, and it is off topic in this thread.

**Comment by Jibun-no-Kage** [comment]
Oh... sorry, did not realize.
